src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/ProjectSystemCache.cs

```diff
@@ -547,5 +548,74 @@ public void Dispose()
             Dispose(true);
             GC.SuppressFinalize(this);
         }
+
+        public bool AddProjectRestoreInfoSource(ProjectNames projectNames, object restoreInfoSource)
+        {
+            if (projectNames == null)
```

Comment:
Should we enable nullable ? and change `ProjectNames `to `ProjectNames?`

Reply:
I think enabling nullable is something on our radar, but at this point it'd add more work than we're willing to do.